test/new_relic/agent/instrumentation/rails/action_view_subscriber.rb

```diff
@@ -110,6 +110,23 @@ def test_records_metrics_for_simple_collection
     assert_metrics_recorded('View/model/_user.html.erb/Partial' => expected)
   end
 
+  def test_records_metrics_for_simple_layout
+    params = {:identifier => '/root/app/views/model/_layout.html.erb'}
+    nr_freeze_process_time
+    in_transaction do
+      @subscriber.start('render_layout.action_view', :id, params)
+      @subscriber.start('!render_layout.action_view', :id,
+        :virtual_path => 'model/_layout')
+      advance_process_time(2.0)
+      @subscriber.finish('!render_layout.action_view', :id,
+        :virtual_path => 'model/_layout')
+      @subscriber.finish('render_layout.action_view', :id, params)
+    end
+    expected = {:call_count => 1, :total_call_time => 2.0}
+
+    assert_metrics_recorded('View/model/_layout.html.erb/Layout' => expected)
+  end
```

Comment:
Could you walk me through this test? What I think I'm seeing is:
* pause the clock
* start two similar events, but one with a false name (because it leads with a `!`), for the same params
* advance the clock
* finish both events
* make sure only one of the events is recorded

Is that right, or is there something else going on too? Are only metrics supposed to be recorded for `render_layout`, not segments?

(It looks like a great test, I just want to make sure I'm reading it correctly!)

Reply:
We chatted about this and the description above matches my understanding, except the events that start with `!` are the ones we want recorded! Descriptive comment available in [NewRelic::Agent::ActionViewSubscriber#recordable?](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/instrumentation/action_view_subscriber.rb#L51-L59)